shell/packages/sandstorm-db/migrations.js

```diff
@@ -358,6 +358,20 @@ function splitAccountUsersAndIdentityUsers() {
   });
 }
 
+function populateContactsFromApiTokens() {
+  ApiTokens.find({"owner.user": {$exists: 1}}).forEach(function(token) {
+    var identityId = token.owner.user.identityId;
+    var profile = SandstormDb.prototype.getIdentity(identityId).profile;
+    Contacts.upsert({ownerId: token.accountId, identityId: identityId}, {
```

Comment:
Note, though, that such tokens are probably very rare. To make one, I think you'd need to have obtained a token from an anonymous offer template and then redeemed it as a /shared/ link as a logged-in user. So I recommend just ignoring them for now.